src/API.jl

```diff
@@ -525,28 +538,33 @@ function gc(ctx::Context=Context(); collect_delay::Period=Day(7), kwargs...)
                 end
             end
         end
-
         merge_orphanages!(Dict(), depot_orphaned_packages, packages_to_delete)
     end
 
+
     # Next, do the same for artifacts.  Note that we MUST do this after calculating
     # `packages_to_delete`, as `process_artifacts_toml()` uses it internally to discount
     # `Artifacts.toml` files that will be deleted by the future culling operation.
     printpkgstyle(ctx, :Active, "artifacts:")
-    artifacts_to_keep = mark(process_artifacts_toml, all_artifacts_files)
+    artifacts_to_keep = mark(x -> process_artifacts_toml(x, packages_to_delete), all_artifacts_files)
+    repos_to_keep = mark(process_manifest_repos, all_manifest_files; do_print=false)
 
-    # Collect all orphaned paths (packages and artifacts that are not reachable).  These
+    # Collect all orphaned paths (packages, artifacts and repos that are not reachable).  These
     # are implicitly defined in that we walk all packages/artifacts installed, then if
     # they were not marked in the above steps, we reap them.
     packages_to_delete = String[]
     artifacts_to_delete = String[]
+    repos_to_delete = String[]
+
     for depot in depots()
         # We track orphaned objects on a per-depot basis, writing out our `orphaned.toml`
         # tracking file immediately, only pushing onto the overall `*_to_delete` lists if
         # the package has been orphaned for at least a period of `collect_delay`
         depot_orphaned_packages = String[]
         depot_orphaned_artifacts = String[]
+        depot_orphaned_repos = String[]
 
+        # ??: This code block is identical to one a bit above
```

Comment:
Hmmm.  You're right.  I think there used to be differences, but they've since been abstracted away.  You can try deleting this block, I think.  I think it will just double-up everything in `depot_orphaned_packages`.